lib/infra/TranscodingEngine.js

```diff
@@ -0,0 +1,64 @@
+
+const Promise = require("bluebird");
+const exec = require('child_process').exec;
+const TranscodingEngineResponse = require('./TranscodingEngineResponse');
+const fs = require('fs');
+const path = require('path');
+
+/**
+ * class to handle transcoding
+ */
+class TranscodingEngine
+{
+
+    constructor(ffmpegPath = 'ffmpeg')
+    {
+        this._ffmpegPath=ffmpegPath;
+    }
+
+    static validateFilePaths(sourceFilePath , pathToSave)
+    {
+        return new Promise(function(resolve,reject) {

```

Comment:
You must use the eslintrc since there are lots of not correctly placed parentheses - please fix all